src/components/VerificationResult.jsx

```diff
@@ -0,0 +1,69 @@
+import React, {useEffect, useState} from 'react';
+import {verify} from "../utils/verification-utils.js";
+import VerificationSuccess from "./VerificationSuccess";
+import VerificationFailed from "./VerificationFailed";
+import { CircularProgress, Grid, Typography} from "@mui/material";
+
+const decompressData = (compressedData) => {
+    //TODO: implement this
+    let decompressedData = compressedData ?? "{}";
+    return decompressedData;
+}
+
+const getVcFromQrData = async (qrData) => {
+    return JSON.parse(decompressData(qrData?.text));
+}
+
+const getVcStatus = (qrData, setVc, setVcStatus, setLoading) => {
+    if (!qrData?.text) return null;
+    getVcFromQrData(qrData).then(vc => {
+        verify(vc)
+            .then(response => {
+                setVcStatus(response);
+                setLoading(false);
+            })
+            .catch(error => {
+                console.error("Error occurred while verifying the VC. Error: ", error);
+                setVcStatus({status: "NOK", checks: []});
```

Comment:
what is "NOK" status?

Reply:
NOK means 'Not OK'. The verification sdk returns status as NOK incase the vc isn't valid. So, setting the same value in case of errors